src/lib.rs

```diff
@@ -186,7 +186,7 @@ macro_rules! test_println {
 mod clear;
 pub mod implementation;
 mod page;
-pub mod pool;
+pub(crate) mod pool;
```

Comment:
I _believe_ this can just be 
```suggestion
mod pool;
```
AFAIK, no other modules in the crate  need to reference the `pool` module's internals, and we're reexporting it from the root, so if they need to in the future, they can just refer to it as `crate::Pool`.